lib/matplotlib/collections.py

```diff
@@ -1254,6 +1255,139 @@ def set_verts_and_codes(self, verts, codes):
         self.stale = True
 
 
+class FillBetweenPolyCollection(PolyCollection):
+    def __init__(
+            self, ind_dir, ind, dep1, dep2=0, *,
+            where=None, interpolate=False, step=None, _axes=None, **kwargs):
+        self.ind_dir = ind_dir
+        axes = _axes or getattr(self, "axes", None)
```

Comment:
e112d5a84e352a774c798bed841465a21ee88d16

Reply:
Unfortunately I have to change it back at this place in 38cd7b9214c1813281f931b72e572b1b5edd2cf7. Reason is that the existence of `self.axes` is only established _after_ `Artist.__init__`, which is implemented at the end of the current `__init__`.

`Artist.__init__` is called so late, because before the initialiser of the super class `PolyCollection` can be called, the `polys` needs to be obtained, which already asks for an `axes`.

Reply:
Argh, the logic is really scewd up here. AFAICS `self.axes` should always exist because it's a property of `Artist`. However, it reads `self._axes`, which is not initialized until after `Artist.__init__`. `getattr(self, "axes", None)` will always return None, not because self.axes does not exist, but because `self.axes` raises an AttributeError on missing self._axes and then `getattr` falls back to the default. Therefore, your code should be equivalent to

```suggestion
        axes = _axes
```

I also suspect that some cases may not work when `_axes` is not passed (i.e. None). The clean solution would be to remove the `axes` dependency at creation time. This would need closer investigation. The hacky way would be to always require `_axes` and pre-populate `self._axes` with it (needs clear comment/documentation as a workaround).

Reply:
- 50552b59732745c7225261fbb9489918f3372038
- Further details will be inspected in https://github.com/matplotlib/matplotlib/pull/28702#discussion_r1745245591